### PR TITLE
Set discussion_url at before_create.

### PR DESCRIPTION
**Pull request by pedrosino**
See https://github.com/cubing/worldcubeassociation.org/pull/715#discussion_r66649526

### CODE REVIEW

File: WcaOnRails/app/controllers/delegate_reports_controller.rb [hunk 1]
```diff
@@ -29,13 +29,14 @@ def update
...
     @delegate_report.current_user = current_user
-    @delegate_report.discussion_url = "https://groups.google.com/forum/#!topicsearchin/wca-delegates/" + URI.encode(@delegate_report.competition.name)
     was_posted = @delegate_report.posted?
     if @delegate_report.update_attributes(delegate_report_params)
       flash[:success] = "Updated report"
       if @delegate_report.posted? && !was_posted
         # Don't email when posting old delegate reports.
         # See https://github.com/cubing/worldcubeassociation.org/issues/704 for details.
+        # Set the discussion_url column only at posting
+        @delegate_report.discussion_url = "https://groups.google.com/forum/#!topicsearchin/wca-delegates/" + URI.encode(@delegate_report.competition.name)
```
Comment (jonatanklosko) [inline WcaOnRails/app/controllers/delegate_reports_controller.rb]: I would suggest to move this logic to the model by wrapping this in a `before_create` callback.

File: WcaOnRails/spec/controllers/delegate_reports_controller_spec.rb [hunk 2]
```diff
@@ -81,19 +81,15 @@
...
+      # Check if the discussion_url was set
+      expect(comp.delegate_report.discussion_url).to eq "https://groups.google.com/forum/#!topicsearchin/wca-delegates/" + URI.encode(comp.name)
 
       # Try to update the report when it's posted.
       post :update, competition_id: comp.id, delegate_report: { remarks: "My newerer remarks" }
       comp.reload
       expect(comp.delegate_report.remarks).to eq "My newer remarks"
     end
 
-    it "updating report sets the discussion_url" do
```
Comment (jfly) [inline WcaOnRails/spec/controllers/delegate_reports_controller_spec.rb]: One idea would be to change this test to say "does set update discussion_url" when updating report.

File: WcaOnRails/app/controllers/delegate_reports_controller.rb (hunk 1 above)
Comment (larspetrus) [inline WcaOnRails/app/controllers/delegate_reports_controller.rb]: This placement murders the comment in lines 36-37.

File: WcaOnRails/app/controllers/delegate_reports_controller.rb [hunk 3]
```diff
@@ -29,13 +29,14 @@ def update
...
     @delegate_report = @competition.delegate_report
     @delegate_report.current_user = current_user
-    @delegate_report.discussion_url = "https://groups.google.com/forum/#!topicsearchin/wca-delegates/" + URI.encode(@delegate_report.competition.name)
     was_posted = @delegate_report.posted?
     if @delegate_report.update_attributes(delegate_report_params)
       flash[:success] = "Updated report"
       if @delegate_report.posted? && !was_posted
         # Don't email when posting old delegate reports.
         # See https://github.com/cubing/worldcubeassociation.org/issues/704 for details.
+        # Set the discussion_url column only at posting
```
Comment (larspetrus) [inline WcaOnRails/app/controllers/delegate_reports_controller.rb]: This comment only explains the obvious. I'd remove it.

File: WcaOnRails/app/models/delegate_report.rb [hunk 4]
```diff
@@ -6,6 +6,11 @@ class DelegateReport < ActiveRecord::Base
 
   attr_accessor :current_user
 
+  before_create :set_discussion_url
+  def set_discussion_url
+    self.discussion_url = "https://groups.google.com/forum/#!topicsearchin/wca-delegates/" + URI.encode(competition.name)
+  end
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/delegate_report.rb]: Actually I've just relised that the best place for this is the [delegate_report method in the Competition model](https://github.com/cubing/worldcubeassociation.org/blob/master/WcaOnRails/app/models/competition.rb#L587-L598).
(Or we can set the `equipment` template here, just be consistent)

**Comment by jonatanklosko** [comment]
LGTM!

**Pull request merged**